src/basics.md

```diff
@@ -277,6 +279,52 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of alphanumeric characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+fn main() {
+    let rand_string: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", rand_string);
+}
+```
+
+[ex-rand-choose]: #ex-rand-choose
+<a name="ex-rand-choose"></a>
+## Create random passwords from a set of user-defined characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters with custom user-defined bytestring, with [`choose`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+fn main() {
+    const CHARSET: &[u8] =  b"ABCDEFGHIJKLMNOPQRSTUVWXYZ\
+    abcdefghijklmnopqrstuvwxyz\
+    0123456789)(*&^%$#@!~";
+
+    let mut rng = thread_rng();
+    let password: String = (0..30)
+        .map(|_| *rng.choose(CHARSET).unwrap() as char)
```

Comment:
we generally avoid `unwrap` in the examples, we should either document why the `unwrap` will be always ok as long as the `CHARSET` is nonempty or provide a generalized solution like:
```rust
    let password: Option<String> = (0..30)
        .map(|_| Some(*rng.choose(CHARSET)? as char))
        .collect();

    println!("{:?}", password);
```

Reply:
@budziq Thanks for showing me how to implement the generator case!  I have updated the PR as suggested to remove the `unwrap()` and print an `Option` instead.